### PR TITLE
add weather page

### PR DESCRIPTION
**Pull request by aaaleeenaaa**

### CODE REVIEW

File: pages/weather.js [hunk 1]
```diff
@@ -1,3 +1,244 @@
...
+
 export default function WeatherPage() {
-  return <h1>Weather page</h1>;
+  const [weatherLocation, setWeatherLocation] = useLocalStorageState(
+    "weatherLocation",
+    { defaultValue: "Hamburg" }
+  );
+  const [url, setUrl] = useState(
+    `https://api.weatherapi.com/v1/forecast.json?key=a7f12ff76d6c4b7383f102231231505&q=${weatherLocation}&days=7&aqi=no&alerts=no`
+  );
```
Comment (NikRoe) [inline pages/weather.js]: this appears like a derived state from the weatherLocation and you can write it like this:

```js
const url = `https://api.weatherapi.com/v1/forecast.json?key=a7f12ff76d6c4b7383f102231231505&q=${weatherLocation}&days=7&aqi=no&alerts=no`;
```

File: pages/weather.js [hunk 2]
```diff
@@ -1,3 +1,244 @@
...
+
+  const { data, error, isLoading } = useSWR(url, fetcher);
+
+  useEffect(() => {
+    setUrl(
+      `https://api.weatherapi.com/v1/forecast.json?key=a7f12ff76d6c4b7383f102231231505&q=${weatherLocation}&days=7&aqi=no&alerts=no`
+    );
+  }, [weatherLocation]);
```
Comment (NikRoe) [inline pages/weather.js]: you don't need a useEffect here and can remove it:

```suggestion
```

File: pages/weather.js [hunk 3]
```diff
@@ -1,3 +1,244 @@
...
+  const todayWeekday = currentDate.toLocaleDateString("en-US", options);
+  const tomorrowWeekday = tomorrowDate.toLocaleDateString("en-US", options);
+
+  function handleKeyPress(event) {
+    if (event.key === "Enter") {
+      event.preventDefault();
+      const newLocation = event.target.value;
+      setWeatherLocation(newLocation);
+    }
+  }
```
Comment (NikRoe) [inline pages/weather.js]: you don't need this function, when hitting "enter" the form is submitted and the handleLocationSubmit is called.

You would set the state twice.

```suggestion
```

File: pages/weather.js [hunk 4]
```diff
@@ -1,3 +1,244 @@
...
+    <StyledSection>
+      <form onSubmit={handleLocationSubmit}>
+        <Label htmlFor="weatherLocation">City: {""}</Label>
+        <Input
+          id="weatherLocation"
+          name="weatherLocation"
+          type="text"
+          maxLength="30"
+          minLength="3"
+          onKeyPress={handleKeyPress}
```
Comment (NikRoe) [inline pages/weather.js]:
```suggestion
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi Alena,

this looks really nice so far 👏 

You can remove a few lines of code to make your code even more readable, while maintaining the same functionality it had before. I left you a few comments regarding this :) 

And I noticed that your API key is exposed and can be read via the Network Tab.
This happens when we fetch from the frontend and provide the API key within the URL we want to fetch.

We can prevent this by making this fetch from the server/backend, for instance via an API proxy.

In your project you would have a new API Route (for instance `/api/weather.js`) where you would fetch the weather API and send the response as json back to the frontend. (Keep in mind that hooks, such as useSWR don't work on the server, so you will need to work with the good old fetch API to fetch from your server)

In your frontend (in `/pages/weather.js`) you would no longer fetch the weather API directly, but fetch your own API Route instead. If you want to provide more information, such as the weatherLocation to fetch, you can do so via the URL).

This does sound like a lot of work, but it will make your App a lot saver and we never want our keys, secrets etc. to be exposed.

Let me know if you need help implementing this. Though I hope my comment already helps :)

**Review by NikRoe — APPROVED** [review]
Hi Alena,

me again 😅 

Really great implementation 🥳 

In your frontend (`/pages/weather.js`) you can still work with useSWR if you like to do so.
The restriction I was describing in my last review was only referring to the server (`/api/weather.js`), I might not have been stating this clearly enough.

Both solutions work and you go with whatever you like more

**Comment by aaaleeenaaa** [comment]
Hi Niklas, thanks for your review and your hint about the useSWR. I guess as it works and as you say it is fine as well, I will just leave it like that now 😅

**Pull request merged**